src/System.Device.Gpio/Interop/Unix/libgpiod/Interop.libgpiod.cs

```diff
@@ -0,0 +1,139 @@
+   /// <param name="chip">The GPIO chip handle</param>
+   /// <param name="offset">The offset of the GPIO line</param>
+   /// <returns>Handle to the GPIO line or NULL if an error occured.</returns>
+   [DllImport(LibgpiodLibrary, SetLastError = true)]
+   internal static extern SafeLineHandle GetChipLineByOffset(SafeChipHandle chip, int offset);
+
+   /// <summary>
+   /// Read the GPIO line direction setting.
+   /// </summary>
+   /// <param name="line">GPIO line handle</param>
+   /// <returns>GPIOD_DIRECTION_INPUT or GPIOD_DIRECTION_OUTPUT.</returns>
+   [DllImport(LibgpiodLibrary)]
+   internal static extern int GetLineDirection(SafeLineHandle line);
+
+   /// <summary>
+   /// Reserve a single line, set the direction to input.
+   /// </summary>
+   /// <param name="line">GPIO line handle</param>
+   /// <param name="consumer">Name of the consumer.</param>
+   /// <returns>0 if the line was properly reserved, -1 on failure.</returns>
+   [DllImport(LibgpiodLibrary)]
+   internal static extern int RequestLineInput(SafeLineHandle line, string consumer);
+
+   /// <summary>
+   /// Reserve a single line, set the direction to output.
+   /// </summary>
+   /// <param name="line">GPIO line handle</param>
+   /// <param name="consumer">Name of the consumer.</param>
+   /// <returns>0 if the line was properly reserved, -1 on failure.</returns>
+   [DllImport(LibgpiodLibrary)]
+   internal static extern int RequestLineOutput(SafeLineHandle line, string consumer);
+
+   /// <summary>
+   /// Set the value of a single GPIO line.
+   /// </summary>
+   /// <param name="line">GPIO line handle</param>
+   /// <param name="value">New value.</param>
+   /// <returns>0 if the operation succeeds. In case of an error this routine returns -1 and sets the last error number.</returns>
+   [DllImport(LibgpiodLibrary)]
+   internal static extern int SetGpiodLineValue(SafeLineHandle line, int value);
+
+   /// <summary>
+   /// Read current value of a single GPIO line.
+   /// </summary>
+   /// <param name="line">GPIO line handle</param>
+   /// <returns>0 or 1 if the operation succeeds. On error this routine returns -1 and sets the last error number.</returns>
+   [DllImport(LibgpiodLibrary, SetLastError = true)]
+   internal static extern int GetGpiodLineValue(SafeLineHandle line);
+
+   /// <summary>
+   /// Release a previously reserved line.
+   /// </summary>
+   /// <param name="line">GPIO line handle</param>
+   [DllImport(LibgpiodLibrary)]
+   internal static extern void ReleaseGpiodLine(SafeLineHandle lineHandle);
+
+   /// <summary>
+   /// Request all event type notifications on a single line.
+   /// </summary>
+   /// <param name="line">GPIO line handle</param>
+   /// <param name="consumer">Name of the consumer.</param>
+   /// <returns>0 the operation succeeds, -1 on failure.</returns>
+   [DllImport(LibgpiodLibrary, SetLastError = true)]
+   internal static extern int RequestBothEdgeEventForLine(SafeLineHandle line, string consumer);
```

Comment:
Same here, I would probably keep this for the next PR.

Reply:
ok